src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -424,13 +430,16 @@ OMapInnerNode::merge_entry(
         //retire extent
         std::vector<laddr_t> dec_laddrs {l->get_laddr(), r->get_laddr()};
         return dec_ref(oc, dec_laddrs
-	).si_then([this, oc] {
+	).si_then([this, oc, r=std::move(replacement)] {
 	  --(oc.t.get_omap_tree_stats().extents_num_delta);
           if (extent_is_below_min()) {
             return merge_entry_ret(
                    interruptible::ready_future_marker{},
-                   mutation_result_t(mutation_status_t::NEED_MERGE,
-		     std::nullopt, this));
+		   is_btree_root()
+		    ? mutation_result_t(mutation_status_t::NEED_MERGE,
+					std::nullopt, this, r)
+		    : mutation_result_t(mutation_status_t::NEED_MERGE,
+					std::nullopt, this));
```

Comment:
Hm, looks https://github.com/ceph/ceph/blob/f0841d736073fbcaacee20be1500fa08501f775c/src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc#L617-L622 is related which is missing the `is_btree_root()` logic?

Reply:
Um, it seems that this may not be necessary, as the `root` of leaf nodes are set by `OMapInnerNode` when doing `merge_entry`.

Reply:
Ah, the leaf node cannot be merged if it is the root.